usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -716,10 +720,21 @@ static CK_RV ep11tok_pkey_wrap_handler(uint_32 adapter, uint_32 domain,
         goto done;
 
     /* Create the protected key via CKM_IBM_CPACF_WRAP */
+repeat:
     ret = dll_m_WrapKey(data->secure_key, data->secure_key_len,
                         NULL, 0, NULL, 0, &mech,
                         data->pkey_buf, data->pkey_buflen_p,
                         target | XCP_TGTFL_SET_SCMD);
+    if (ret == CKR_SESSION_CLOSED && retry == FALSE &&
+        data->ep11_session != NULL) {
```

Comment:
Is this a realistic scenario, retrying after a CKR_SESSION_CLOSED? Why should the session then be open? Would it make sense to add a small delay before the retry?

Reply:
Usually this happens when an APQN just got online, and thus have not yet logged in the session (because the login was performed some time before). 
So it can happen that the APQN just got online before ep11tok_pkey_skey2pkey() got called. So handle_all_ep11_cards() will call ep11tok_pkey_wrap_handler() for that APQN, but it might not yet have the session logged in. 
So yes, it is realistic that this can happen, although it is not very likely.